### ISSUE TITLE
1.4.1 relative snapshot location cannot be in parent directory

### ISSUE THREAD

**Issue by tomsmalley** [commercialhaskell/stack#3263]
Version 1.4.1, Git revision 84114192b920a207cdcb12f5498fee2b83b2b85e x86_64 hpack-0.17.0

Relative paths seem to work:

```yaml
resolver: snapshot.yaml
```

Unless they traverse to a parent directory:

```yaml
resolver: ../snapshot.yaml
```

It seems to be attempting to parse as an absolute directory by just appending the resolver location to the current path. Of course using a full absolute path works fine.

```
Could not parse '/Users/Tom/Projects/example/client/stack.yaml':
Aeson exception:
Error in $: InvalidAbsFile "/Users/Tom/Projects/example/client/../snapshot.yaml"
See http://docs.haskellstack.org/en/stable/yaml_configuration/
```

**Comment by snoyberg** [comment]
Thanks for the report, good catch. This should be fixed with 20dc5ef04df20171e8f7b21303bed4e00873223f.

**Comment by tomsmalley** [comment]
Seems to work! Many thanks.
